<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>A Supervised Approach for Predicting Cell Cycle Progression Using
    Single-Cell RNA-seq Data • peco</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="A Supervised Approach for Predicting Cell Cycle Progression Using
    Single-Cell RNA-seq Data">
<meta property="og:description" content="Our methods provide a way to assign continuous cell cycle
    phase using single-cell RNA-seq data, and consequently, allow for
    identication cyclic trend of gene expression levels along the cell
    cycle. This package provides method and training data, including
    single-cell RNA-seq data collected from 6 individual cell lines of
    induced pluripotent stem cells (iPSCs), and also continuous cell
    cycle phase derived from FUCCI fluorescence imaging
    data. cycle_npreg_insample and cycle_npreg_outsample are the two
    main functions in peco. cycle_npreg_insample generates cyclic
    trend estimates of gene expression levels using training data, and
    cycle_npreg_outsample applies the estimates of
    cycle_npreg_insample to another gene expression dataset to infer
    an angle or cell cycle phase for each cell.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">peco</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.15</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="articles/index.html">Vignettes</a>
</li>
<li>
  <a href="reference/index.html">Functions</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jhsiao999/peco">Source</a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">


<div id="peco" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#peco" class="anchor"></a>peco</h1></div>
<p><strong>peco</strong> is an R package for <strong>P</strong>r<strong>E</strong>dicting <strong>C</strong>ell cycle pr<strong>O</strong>gression in a continuum using scRNA-seq data.</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To install and load the package, run:</p>
<div class="sourceCode" id="cb1"><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"devtools"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">devtools</span>)
<span class="fu"><a href="https://rdrr.io/pkg/devtools/man/remote-reexports.html">install_github</a></span>(<span class="st">"jhsiao999/peco"</span>)</pre></div>
<p><code>peco</code> uses <code>SingleCellExperiment</code> class objects.</p>
<div class="sourceCode" id="cb2"><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">peco</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">SingleCellExperiment</span>)</pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="co">## Loading required package: SummarizedExperiment</span></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"></a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">## Loading required package: GenomicRanges</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">## Loading required package: stats4</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">## Loading required package: BiocGenerics</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co">## Loading required package: parallel</span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">## </span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co">## Attaching package: 'BiocGenerics'</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="co">## The following objects are masked from 'package:parallel':</span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="co">## </span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="co">##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="co">##     clusterExport, clusterMap, parApply, parCapply, parLapply,</span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="co">##     parLapplyLB, parRapply, parSapply, parSapplyLB</span></a>
<a class="sourceLine" id="cb3-19" data-line-number="19"></a>
<a class="sourceLine" id="cb3-20" data-line-number="20"><span class="co">## The following objects are masked from 'package:stats':</span></a>
<a class="sourceLine" id="cb3-21" data-line-number="21"><span class="co">## </span></a>
<a class="sourceLine" id="cb3-22" data-line-number="22"><span class="co">##     IQR, mad, sd, var, xtabs</span></a>
<a class="sourceLine" id="cb3-23" data-line-number="23"></a>
<a class="sourceLine" id="cb3-24" data-line-number="24"><span class="co">## The following objects are masked from 'package:base':</span></a>
<a class="sourceLine" id="cb3-25" data-line-number="25"><span class="co">## </span></a>
<a class="sourceLine" id="cb3-26" data-line-number="26"><span class="co">##     anyDuplicated, append, as.data.frame, basename, cbind,</span></a>
<a class="sourceLine" id="cb3-27" data-line-number="27"><span class="co">##     colMeans, colnames, colSums, dirname, do.call, duplicated,</span></a>
<a class="sourceLine" id="cb3-28" data-line-number="28"><span class="co">##     eval, evalq, Filter, Find, get, grep, grepl, intersect,</span></a>
<a class="sourceLine" id="cb3-29" data-line-number="29"><span class="co">##     is.unsorted, lapply, lengths, Map, mapply, match, mget, order,</span></a>
<a class="sourceLine" id="cb3-30" data-line-number="30"><span class="co">##     paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind,</span></a>
<a class="sourceLine" id="cb3-31" data-line-number="31"><span class="co">##     Reduce, rowMeans, rownames, rowSums, sapply, setdiff, sort,</span></a>
<a class="sourceLine" id="cb3-32" data-line-number="32"><span class="co">##     table, tapply, union, unique, unsplit, which, which.max,</span></a>
<a class="sourceLine" id="cb3-33" data-line-number="33"><span class="co">##     which.min</span></a>
<a class="sourceLine" id="cb3-34" data-line-number="34"></a>
<a class="sourceLine" id="cb3-35" data-line-number="35"><span class="co">## Loading required package: S4Vectors</span></a>
<a class="sourceLine" id="cb3-36" data-line-number="36"></a>
<a class="sourceLine" id="cb3-37" data-line-number="37"><span class="co">## </span></a>
<a class="sourceLine" id="cb3-38" data-line-number="38"><span class="co">## Attaching package: 'S4Vectors'</span></a>
<a class="sourceLine" id="cb3-39" data-line-number="39"></a>
<a class="sourceLine" id="cb3-40" data-line-number="40"><span class="co">## The following object is masked from 'package:base':</span></a>
<a class="sourceLine" id="cb3-41" data-line-number="41"><span class="co">## </span></a>
<a class="sourceLine" id="cb3-42" data-line-number="42"><span class="co">##     expand.grid</span></a>
<a class="sourceLine" id="cb3-43" data-line-number="43"></a>
<a class="sourceLine" id="cb3-44" data-line-number="44"><span class="co">## Loading required package: IRanges</span></a>
<a class="sourceLine" id="cb3-45" data-line-number="45"></a>
<a class="sourceLine" id="cb3-46" data-line-number="46"><span class="co">## Loading required package: GenomeInfoDb</span></a>
<a class="sourceLine" id="cb3-47" data-line-number="47"></a>
<a class="sourceLine" id="cb3-48" data-line-number="48"><span class="co">## Loading required package: Biobase</span></a>
<a class="sourceLine" id="cb3-49" data-line-number="49"></a>
<a class="sourceLine" id="cb3-50" data-line-number="50"><span class="co">## Welcome to Bioconductor</span></a>
<a class="sourceLine" id="cb3-51" data-line-number="51"><span class="co">## </span></a>
<a class="sourceLine" id="cb3-52" data-line-number="52"><span class="co">##     Vignettes contain introductory material; view with</span></a>
<a class="sourceLine" id="cb3-53" data-line-number="53"><span class="co">##     'browseVignettes()'. To cite Bioconductor, see</span></a>
<a class="sourceLine" id="cb3-54" data-line-number="54"><span class="co">##     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></a>
<a class="sourceLine" id="cb3-55" data-line-number="55"></a>
<a class="sourceLine" id="cb3-56" data-line-number="56"><span class="co">## Loading required package: DelayedArray</span></a>
<a class="sourceLine" id="cb3-57" data-line-number="57"></a>
<a class="sourceLine" id="cb3-58" data-line-number="58"><span class="co">## Loading required package: matrixStats</span></a>
<a class="sourceLine" id="cb3-59" data-line-number="59"></a>
<a class="sourceLine" id="cb3-60" data-line-number="60"><span class="co">## </span></a>
<a class="sourceLine" id="cb3-61" data-line-number="61"><span class="co">## Attaching package: 'matrixStats'</span></a>
<a class="sourceLine" id="cb3-62" data-line-number="62"></a>
<a class="sourceLine" id="cb3-63" data-line-number="63"><span class="co">## The following objects are masked from 'package:Biobase':</span></a>
<a class="sourceLine" id="cb3-64" data-line-number="64"><span class="co">## </span></a>
<a class="sourceLine" id="cb3-65" data-line-number="65"><span class="co">##     anyMissing, rowMedians</span></a>
<a class="sourceLine" id="cb3-66" data-line-number="66"></a>
<a class="sourceLine" id="cb3-67" data-line-number="67"><span class="co">## Loading required package: BiocParallel</span></a>
<a class="sourceLine" id="cb3-68" data-line-number="68"></a>
<a class="sourceLine" id="cb3-69" data-line-number="69"><span class="co">## </span></a>
<a class="sourceLine" id="cb3-70" data-line-number="70"><span class="co">## Attaching package: 'DelayedArray'</span></a>
<a class="sourceLine" id="cb3-71" data-line-number="71"></a>
<a class="sourceLine" id="cb3-72" data-line-number="72"><span class="co">## The following objects are masked from 'package:matrixStats':</span></a>
<a class="sourceLine" id="cb3-73" data-line-number="73"><span class="co">## </span></a>
<a class="sourceLine" id="cb3-74" data-line-number="74"><span class="co">##     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</span></a>
<a class="sourceLine" id="cb3-75" data-line-number="75"></a>
<a class="sourceLine" id="cb3-76" data-line-number="76"><span class="co">## The following objects are masked from 'package:base':</span></a>
<a class="sourceLine" id="cb3-77" data-line-number="77"><span class="co">## </span></a>
<a class="sourceLine" id="cb3-78" data-line-number="78"><span class="co">##     aperm, apply</span></a></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">doParallel</span>)</pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="co">## Loading required package: foreach</span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="co">## Loading required package: iterators</span></a></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">foreach</span>)</pre></div>
</div>
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p><code>peco</code> is a supervised approach for PrEdicting cell cycle phase in a COntinuum using single-cell RNA sequencing data. The R package provides functions to build training dataset and also functions to use existing training data to predict cell cycle on a continuum.</p>
<p>Our work demonstrated that peco is able to predict continuous cell cylce phase using a small set of cylcic genes: <em>CDK1</em>, <em>UBE2C</em>, <em>TOP2A</em>, <em>HISTH1E</em>, and <em>HISTH1C</em> (identified as cell cycle marker genes in studies of yeast (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC25624">Spellman et al., 1998</a>) and HeLa cells (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC117619">Whitfield et al., 2002</a>)).</p>
<p>Below we provide two use cases. Vignette 1 shows how to use the built-training dataset to predict continuous cell cycle. Vignette 2 shows how to make a training datast and build a predictor using training data.</p>
<p>Users can also view the vigenettes via <code><a href="https://rdrr.io/r/utils/browseVignettes.html">browseVignettes("peco")</a></code>.</p>
</div>
<div id="about-the-training-dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#about-the-training-dataset" class="anchor"></a>About the training dataset</h2>
<p><code>training_human</code> stores built-in training data of 101 significant cyclic genes. Below are the slots contained in <code>training_human</code>:</p>
<ul>
<li>
<code>predict.yy</code>: a gene by sample matrix (101 by 888) that stores predict cyclic expression values.</li>
<li>
<code>cellcycle_peco_reordered</code>: cell cycle phase in a unit circle (angle), ordered from 0 to 2<span class="math inline"><em>p</em><em>i</em></span>
</li>
<li>
<code>cellcycle_function</code>: lists of 101 function corresponding to the top 101 cyclic genes identified in our dataset</li>
<li>
<code>sigma</code>: standard error associated with cyclic trends of gene expression</li>
<li>
<code>pve</code>: proportion of variance explained by the cyclic trend</li>
</ul>
<!-- end list --><div class="sourceCode" id="cb7"><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"training_human"</span>)</pre></div>
</div>
<div id="predict-cell-cycle-phase-using-gene-expression-data" class="section level2">
<h2 class="hasAnchor">
<a href="#predict-cell-cycle-phase-using-gene-expression-data" class="anchor"></a>Predict cell cycle phase using gene expression data</h2>
<p><code>peco</code> is integrated with <code>SingleCellExperiment</code> object in Bioconductor. Below shows an example of inputting <code>SingleCellExperiment</code> object to perform cell cycle phase prediction.</p>
<p><code>sce_top101genes</code> includes 101 genes and 888 single-cell samples and one assay slot of <code>counts</code>.</p>
<div class="sourceCode" id="cb8"><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"sce_top101genes"</span>)
<span class="fu">assays</span>(<span class="no">sce_top101genes</span>)</pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="co">## List of length 1</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2"><span class="co">## names(1): counts</span></a></code></pre></div>
<p>Transform the expression values to quantile-normalizesd counts-per-million values. <code>peco</code> uses the <code>cpm_quantNormed</code> slot as input data for predictions.</p>
<div class="sourceCode" id="cb10"><pre class="r"><span class="no">sce_top101genes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="reference/data_transform_quantile.html">data_transform_quantile</a></span>(<span class="no">sce_top101genes</span>)</pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="co">## computing on 2 cores</span></a></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="r"><span class="fu">assays</span>(<span class="no">sce_top101genes</span>)</pre></div>
<div class="sourceCode" id="cb13"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1"><span class="co">## List of length 3</span></a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="co">## names(3): counts cpm cpm_quantNormed</span></a></code></pre></div>
<p>Apply the prediction model using function <code>cycle_npreg_outsample</code> and generate prediction results contained in a list object <code>pred_top101genes</code>.</p>
<div class="sourceCode" id="cb14"><pre class="r"><span class="no">pred_top101genes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="reference/cycle_npreg_outsample.html">cycle_npreg_outsample</a></span>(
    <span class="kw">Y_test</span><span class="kw">=</span><span class="no">sce_top101genes</span>,
    <span class="kw">sigma_est</span><span class="kw">=</span><span class="no">training_human</span>$<span class="no">sigma</span>[<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">sce_top101genes</span>),],
    <span class="kw">funs_est</span><span class="kw">=</span><span class="no">training_human</span>$<span class="no">cellcycle_function</span>[<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">sce_top101genes</span>)],
    <span class="kw">method.trend</span><span class="kw">=</span><span class="st">"trendfilter"</span>,
    <span class="kw">ncores</span><span class="kw">=</span><span class="fl">1</span>,
    <span class="kw">get_trend_estimates</span><span class="kw">=</span><span class="fl">FALSE</span>)</pre></div>
<p>The <code>pred_top101genes$Y</code> contains a SingleCellExperiment object with the predict cell cycle phase in the <code>colData</code> slot.</p>
<div class="sourceCode" id="cb15"><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">colData</span>(<span class="no">pred_top101genes</span>$<span class="no">Y</span>)$<span class="no">cellcycle_peco</span>)</pre></div>
<div class="sourceCode" id="cb16"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1"><span class="co">## 20170905-A01 20170905-A02 20170905-A03 20170905-A06 20170905-A07 </span></a>
<a class="sourceLine" id="cb16-2" data-line-number="2"><span class="co">##     1.099557     4.555309     2.481858     4.303982     4.052655 </span></a>
<a class="sourceLine" id="cb16-3" data-line-number="3"><span class="co">## 20170905-A08 </span></a>
<a class="sourceLine" id="cb16-4" data-line-number="4"><span class="co">##     1.413717</span></a></code></pre></div>
<p>Visualize results of prediction for one gene. Below we choose CDK1 (“ENSG00000170312”). Because CDK1 is a known cell cycle gene, this visualization serves as a sanity check for the results of fitting. The fitted function <code>training_human$cellcycle_function[[1]]</code> was obtained from our training data.</p>
<div class="sourceCode" id="cb17"><pre class="r"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">y</span><span class="kw">=</span><span class="fu">assay</span>(<span class="no">pred_top101genes</span>$<span class="no">Y</span>,<span class="st">"cpm_quantNormed"</span>)[<span class="st">"ENSG00000170312"</span>,],
     <span class="kw">x</span><span class="kw">=</span><span class="fu">colData</span>(<span class="no">pred_top101genes</span>$<span class="no">Y</span>)$<span class="no">theta_shifted</span>, <span class="kw">main</span> <span class="kw">=</span> <span class="st">"CDK1"</span>,
     <span class="kw">ylab</span> <span class="kw">=</span> <span class="st">"quantile normalized expression"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="kw">y</span><span class="kw">=</span><span class="no">training_human</span>$<span class="no">cellcycle_function</span><span class="kw">[[</span><span class="st">"ENSG00000170312"</span>]](<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">0</span>,<span class="fl">2</span>*<span class="no">pi</span>, <span class="kw">length.out</span><span class="kw">=</span><span class="fl">100</span>)),
       <span class="kw">x</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">0</span>,<span class="fl">2</span>*<span class="no">pi</span>, <span class="kw">length.out</span><span class="kw">=</span><span class="fl">100</span>), <span class="kw">col</span> <span class="kw">=</span> <span class="st">"blue"</span>, <span class="kw">pch</span> <span class="kw">=</span><span class="fl">16</span>)</pre></div>
<p><img src="README_files/figure-gfm/unnamed-chunk-7-1.png"><!-- --></p>
</div>
<div id="visualize-cyclic-expression-trend-based-on-predicted-phase" class="section level2">
<h2 class="hasAnchor">
<a href="#visualize-cyclic-expression-trend-based-on-predicted-phase" class="anchor"></a>Visualize cyclic expression trend based on predicted phase</h2>
<p>Visualize results of prediction for the top 10 genesone genes. Use <code>fit_cyclical_many</code> to estimate cyclic function based on the input data.</p>
<div class="sourceCode" id="cb18"><pre class="r"><span class="co"># predicted cell time in the input data</span>
<span class="no">theta_predict</span> <span class="kw">=</span> <span class="fu">colData</span>(<span class="no">pred_top101genes</span>$<span class="no">Y</span>)$<span class="no">cellcycle_peco</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">theta_predict</span>) <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="fu">colData</span>(<span class="no">pred_top101genes</span>$<span class="no">Y</span>))

<span class="co"># expression values of 10 genes in the input data</span>
<span class="no">yy_input</span> <span class="kw">=</span> <span class="fu">assay</span>(<span class="no">pred_top101genes</span>$<span class="no">Y</span>,<span class="st">"cpm_quantNormed"</span>)[<span class="fl">1</span>:<span class="fl">6</span>,]

<span class="co"># apply trendfilter to estimate cyclic gene expression trend</span>
<span class="no">fit_cyclic</span> <span class="kw">&lt;-</span> <span class="fu"><a href="reference/fit_cyclical_many.html">fit_cyclical_many</a></span>(<span class="kw">Y</span><span class="kw">=</span><span class="no">yy_input</span>,
                                <span class="kw">theta</span><span class="kw">=</span><span class="no">theta_predict</span>)</pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="co">## computing on 2 cores</span></a></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="r"><span class="no">gene_symbols</span> <span class="kw">=</span> <span class="fu">rowData</span>(<span class="no">pred_top101genes</span>$<span class="no">Y</span>)$<span class="no">hgnc</span>[<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">yy_input</span>)]

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="kw">mfrow</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">2</span>,<span class="fl">3</span>))
<span class="kw">for</span> (<span class="no">i</span> <span class="kw">in</span> <span class="fl">1</span>:<span class="fl">6</span>) {
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">y</span><span class="kw">=</span><span class="no">yy_input</span>[<span class="no">i</span>,],
     <span class="kw">x</span><span class="kw">=</span><span class="no">fit_cyclic</span>$<span class="no">cellcycle_peco_ordered</span>,
     <span class="kw">main</span> <span class="kw">=</span> <span class="no">gene_symbols</span>[<span class="no">i</span>],
     <span class="kw">ylab</span> <span class="kw">=</span> <span class="st">"quantile normalized expression"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="kw">y</span><span class="kw">=</span><span class="no">fit_cyclic</span>$<span class="no">cellcycle_function</span><span class="kw">[[</span><span class="no">i</span>]](<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">0</span>,<span class="fl">2</span>*<span class="no">pi</span>, <span class="kw">length.out</span><span class="kw">=</span><span class="fl">100</span>)),
       <span class="kw">x</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">0</span>,<span class="fl">2</span>*<span class="no">pi</span>, <span class="kw">length.out</span><span class="kw">=</span><span class="fl">100</span>), <span class="kw">col</span> <span class="kw">=</span> <span class="st">"blue"</span>, <span class="kw">pch</span> <span class="kw">=</span><span class="fl">16</span>)
}</pre></div>
<p><img src="README_files/figure-gfm/unnamed-chunk-8-1.png"><!-- --></p>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h2>
<div class="sourceCode" id="cb21"><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</pre></div>
<div class="sourceCode" id="cb22"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1"><span class="co">## R version 3.5.1 (2018-07-02)</span></a>
<a class="sourceLine" id="cb22-2" data-line-number="2"><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></a>
<a class="sourceLine" id="cb22-3" data-line-number="3"><span class="co">## Running under: Scientific Linux 7.4 (Nitrogen)</span></a>
<a class="sourceLine" id="cb22-4" data-line-number="4"><span class="co">## </span></a>
<a class="sourceLine" id="cb22-5" data-line-number="5"><span class="co">## Matrix products: default</span></a>
<a class="sourceLine" id="cb22-6" data-line-number="6"><span class="co">## BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so</span></a>
<a class="sourceLine" id="cb22-7" data-line-number="7"><span class="co">## </span></a>
<a class="sourceLine" id="cb22-8" data-line-number="8"><span class="co">## locale:</span></a>
<a class="sourceLine" id="cb22-9" data-line-number="9"><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></a>
<a class="sourceLine" id="cb22-10" data-line-number="10"><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></a>
<a class="sourceLine" id="cb22-11" data-line-number="11"><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></a>
<a class="sourceLine" id="cb22-12" data-line-number="12"><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></a>
<a class="sourceLine" id="cb22-13" data-line-number="13"><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></a>
<a class="sourceLine" id="cb22-14" data-line-number="14"><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></a>
<a class="sourceLine" id="cb22-15" data-line-number="15"><span class="co">## </span></a>
<a class="sourceLine" id="cb22-16" data-line-number="16"><span class="co">## attached base packages:</span></a>
<a class="sourceLine" id="cb22-17" data-line-number="17"><span class="co">## [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span></a>
<a class="sourceLine" id="cb22-18" data-line-number="18"><span class="co">## [8] methods   base     </span></a>
<a class="sourceLine" id="cb22-19" data-line-number="19"><span class="co">## </span></a>
<a class="sourceLine" id="cb22-20" data-line-number="20"><span class="co">## other attached packages:</span></a>
<a class="sourceLine" id="cb22-21" data-line-number="21"><span class="co">##  [1] doParallel_1.0.14           iterators_1.0.12           </span></a>
<a class="sourceLine" id="cb22-22" data-line-number="22"><span class="co">##  [3] foreach_1.4.4               SingleCellExperiment_1.4.1 </span></a>
<a class="sourceLine" id="cb22-23" data-line-number="23"><span class="co">##  [5] SummarizedExperiment_1.12.0 DelayedArray_0.8.0         </span></a>
<a class="sourceLine" id="cb22-24" data-line-number="24"><span class="co">##  [7] BiocParallel_1.16.0         matrixStats_0.55.0         </span></a>
<a class="sourceLine" id="cb22-25" data-line-number="25"><span class="co">##  [9] Biobase_2.42.0              GenomicRanges_1.34.0       </span></a>
<a class="sourceLine" id="cb22-26" data-line-number="26"><span class="co">## [11] GenomeInfoDb_1.18.1         IRanges_2.16.0             </span></a>
<a class="sourceLine" id="cb22-27" data-line-number="27"><span class="co">## [13] S4Vectors_0.20.1            BiocGenerics_0.28.0        </span></a>
<a class="sourceLine" id="cb22-28" data-line-number="28"><span class="co">## [15] peco_0.99.10               </span></a>
<a class="sourceLine" id="cb22-29" data-line-number="29"><span class="co">## </span></a>
<a class="sourceLine" id="cb22-30" data-line-number="30"><span class="co">## loaded via a namespace (and not attached):</span></a>
<a class="sourceLine" id="cb22-31" data-line-number="31"><span class="co">##  [1] Rcpp_1.0.3               mvtnorm_1.0-11          </span></a>
<a class="sourceLine" id="cb22-32" data-line-number="32"><span class="co">##  [3] lattice_0.20-38          assertthat_0.2.1        </span></a>
<a class="sourceLine" id="cb22-33" data-line-number="33"><span class="co">##  [5] rprojroot_1.3-2          digest_0.6.20           </span></a>
<a class="sourceLine" id="cb22-34" data-line-number="34"><span class="co">##  [7] plyr_1.8.4               R6_2.4.0                </span></a>
<a class="sourceLine" id="cb22-35" data-line-number="35"><span class="co">##  [9] backports_1.1.2          genlasso_1.4            </span></a>
<a class="sourceLine" id="cb22-36" data-line-number="36"><span class="co">## [11] evaluate_0.12            pracma_2.2.9            </span></a>
<a class="sourceLine" id="cb22-37" data-line-number="37"><span class="co">## [13] ggplot2_3.2.1            pillar_1.3.1            </span></a>
<a class="sourceLine" id="cb22-38" data-line-number="38"><span class="co">## [15] zlibbioc_1.28.0          rlang_0.4.0             </span></a>
<a class="sourceLine" id="cb22-39" data-line-number="39"><span class="co">## [17] lazyeval_0.2.1           Matrix_1.2-17           </span></a>
<a class="sourceLine" id="cb22-40" data-line-number="40"><span class="co">## [19] rmarkdown_1.10           geigen_2.3              </span></a>
<a class="sourceLine" id="cb22-41" data-line-number="41"><span class="co">## [21] stringr_1.3.1            igraph_1.2.2            </span></a>
<a class="sourceLine" id="cb22-42" data-line-number="42"><span class="co">## [23] RCurl_1.95-4.11          munsell_0.5.0           </span></a>
<a class="sourceLine" id="cb22-43" data-line-number="43"><span class="co">## [25] HDF5Array_1.10.1         vipor_0.4.5             </span></a>
<a class="sourceLine" id="cb22-44" data-line-number="44"><span class="co">## [27] compiler_3.5.1           pkgconfig_2.0.3         </span></a>
<a class="sourceLine" id="cb22-45" data-line-number="45"><span class="co">## [29] ggbeeswarm_0.6.0         htmltools_0.3.6         </span></a>
<a class="sourceLine" id="cb22-46" data-line-number="46"><span class="co">## [31] tidyselect_0.2.5         tibble_2.1.1            </span></a>
<a class="sourceLine" id="cb22-47" data-line-number="47"><span class="co">## [33] gridExtra_2.3            GenomeInfoDbData_1.2.0  </span></a>
<a class="sourceLine" id="cb22-48" data-line-number="48"><span class="co">## [35] codetools_0.2-15         viridisLite_0.3.0       </span></a>
<a class="sourceLine" id="cb22-49" data-line-number="49"><span class="co">## [37] crayon_1.3.4             dplyr_0.8.0.1           </span></a>
<a class="sourceLine" id="cb22-50" data-line-number="50"><span class="co">## [39] MASS_7.3-51.1            bitops_1.0-6            </span></a>
<a class="sourceLine" id="cb22-51" data-line-number="51"><span class="co">## [41] grid_3.5.1               gtable_0.2.0            </span></a>
<a class="sourceLine" id="cb22-52" data-line-number="52"><span class="co">## [43] magrittr_1.5             scales_1.0.0            </span></a>
<a class="sourceLine" id="cb22-53" data-line-number="53"><span class="co">## [45] stringi_1.2.4            reshape2_1.4.3          </span></a>
<a class="sourceLine" id="cb22-54" data-line-number="54"><span class="co">## [47] XVector_0.22.0           viridis_0.5.1           </span></a>
<a class="sourceLine" id="cb22-55" data-line-number="55"><span class="co">## [49] scater_1.10.1            DelayedMatrixStats_1.4.0</span></a>
<a class="sourceLine" id="cb22-56" data-line-number="56"><span class="co">## [51] boot_1.3-20              Rhdf5lib_1.4.3          </span></a>
<a class="sourceLine" id="cb22-57" data-line-number="57"><span class="co">## [53] tools_3.5.1              beeswarm_0.2.3          </span></a>
<a class="sourceLine" id="cb22-58" data-line-number="58"><span class="co">## [55] glue_1.3.0               purrr_0.3.2             </span></a>
<a class="sourceLine" id="cb22-59" data-line-number="59"><span class="co">## [57] yaml_2.2.0               colorspace_1.3-2        </span></a>
<a class="sourceLine" id="cb22-60" data-line-number="60"><span class="co">## [59] rhdf5_2.26.2             circular_0.4-93         </span></a>
<a class="sourceLine" id="cb22-61" data-line-number="61"><span class="co">## [61] conicfit_1.0.4           knitr_1.20</span></a></code></pre></div>
<hr>
<div id="contact" class="section level4">
<h4 class="hasAnchor">
<a href="#contact" class="anchor"></a>Contact</h4>
<p>Please contact me at <a href="joyce.hsiao1@gmail.com">joyce.hsiao1@gmail.com</a> for questions on the package or the methods.</p>
</div>
<div id="how-to-cite" class="section level4">
<h4 class="hasAnchor">
<a href="#how-to-cite" class="anchor"></a>How to cite</h4>
<blockquote>
<p>Hsiao, C. J., Tung, P., Blischak, J. D., Burnett, J., Dey, K. K., Barr, A. K., Stephens, M., and Gilad, Y. (2018). <a href="https://doi.org/10.1101/526848">Characterizing and inferring quantitative cell-cycle phase in single-cell RNA-seq data analysis.</a> bioRxiv <a href="doi:10.1101/526848" class="uri">doi:10.1101/526848</a></p>
</blockquote>
</div>
<div id="license" class="section level4">
<h4 class="hasAnchor">
<a href="#license" class="anchor"></a>License</h4>
<p>Copyright (c) 2019-2020, Joyce Hsiao.</p>
<p>All source code and software in this repository are made available under the terms of the <a href="https://www.gnu.org/licenses/gpl-3.0.en.html">GNU General Public License</a>. See file <a href="LICENSE">LICENSE</a> for the full text of the license.</p>
</div>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from BIOC at <br><a href="https://www.bioconductor.org/packages/peco">https://​www.bioconductor.org/​packages/​peco</a>
</li>
<li>Browse source code at <br><a href="https://github.com/jhsiao999/peco/">https://​github.com/​jhsiao999/​peco/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/jhsiao999/peco/issues">https://​github.com/​jhsiao999/​peco/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>GPL (&gt;= 3)</li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing peco</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Chiaowen Joyce Hsiao <br><small class="roles"> Author, maintainer </small>  </li>
<li>Matthew Stephens <br><small class="roles"> Author </small>  </li>
<li><a href="authors.html">All authors...</a></li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://travis-ci.com/jhsiao999/peco"><img src="https://travis-ci.com/jhsiao999/peco.svg?branch=master" alt="Travis-CI Build Status"></a></li>
<li><a href="https://ci.appveyor.com/project/jhsiao999/peco"><img src="https://ci.appveyor.com/api/projects/status/github/jhsiao999/peco?branch=master&amp;svg=true" alt="AppVeyor build status"></a></li>
<li><a href="https://circleci.com/gh/jhsiao999/peco"><img src="https://circleci.com/gh/jhsiao999/peco.svg?style=svg" alt="CircleCI build status"></a></li>
<li><a href="http://www.gnu.org/licenses/gpl-3.0"><img src="https://img.shields.io/badge/License-GPL%20v3-blue.svg" alt="License: GPL v3"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Chiaowen Joyce Hsiao, Matthew Stephens.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
