<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Predicting cell cycle phase using peco • peco</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Predicting cell cycle phase using peco">
<meta property="og:description" content="peco">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">peco</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.12</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../articles/index.html">Vignettes</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jhsiao999/peco">Source</a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Predicting cell cycle phase using peco</h1>
                        <h4 class="author">Joyce Hsiao</h4>
            
            <h4 class="date">2020-11-10</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jhsiao999/peco/blob/master/vignettes/peco.Rmd"><code>vignettes/peco.Rmd</code></a></small>
      <div class="hidden name"><code>peco.Rmd</code></div>

    </div>

    
    
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To install and load the package, run:</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"devtools"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">devtools</span>)
<span class="fu"><a href="https://rdrr.io/pkg/devtools/man/remote-reexports.html">install_github</a></span>(<span class="st">"jhsiao999/peco"</span>)</pre></body></html></div>
<p><code>peco</code> uses <code>SingleCellExperiment</code> class objects.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">SingleCellExperiment</span>)
<span class="co">#&gt; Loading required package: SummarizedExperiment</span>
<span class="co">#&gt; Loading required package: GenomicRanges</span>
<span class="co">#&gt; Loading required package: stats4</span>
<span class="co">#&gt; Loading required package: BiocGenerics</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'BiocGenerics'</span>
<span class="co">#&gt; The following objects are masked from 'package:parallel':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
<span class="co">#&gt;     clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
<span class="co">#&gt;     parLapplyLB, parRapply, parSapply, parSapplyLB</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     IQR, mad, sd, var, xtabs</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
<span class="co">#&gt;     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
<span class="co">#&gt;     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
<span class="co">#&gt;     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
<span class="co">#&gt;     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
<span class="co">#&gt;     union, unique, unsplit, which, which.max, which.min</span>
<span class="co">#&gt; Loading required package: S4Vectors</span>
<span class="co">#&gt; Warning: package 'S4Vectors' was built under R version 3.6.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'S4Vectors'</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand.grid</span>
<span class="co">#&gt; Loading required package: IRanges</span>
<span class="co">#&gt; Loading required package: GenomeInfoDb</span>
<span class="co">#&gt; Warning: package 'GenomeInfoDb' was built under R version 3.6.3</span>
<span class="co">#&gt; Loading required package: Biobase</span>
<span class="co">#&gt; Welcome to Bioconductor</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     Vignettes contain introductory material; view with</span>
<span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="co">#&gt; Loading required package: DelayedArray</span>
<span class="co">#&gt; Warning: package 'DelayedArray' was built under R version 3.6.3</span>
<span class="co">#&gt; Loading required package: matrixStats</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'matrixStats'</span>
<span class="co">#&gt; The following objects are masked from 'package:Biobase':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyMissing, rowMedians</span>
<span class="co">#&gt; Loading required package: BiocParallel</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'DelayedArray'</span>
<span class="co">#&gt; The following objects are masked from 'package:matrixStats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     aperm, apply, rowsum</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">doParallel</span>)
<span class="co">#&gt; Loading required package: foreach</span>
<span class="co">#&gt; Loading required package: iterators</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">foreach</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">peco</span>)</pre></body></html></div>
</div>
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p><code>peco</code> is a supervised approach for PrEdicting cell cycle phase in a COntinuum using single-cell RNA sequencing data. The R package provides functions to build training dataset and also functions to use existing training data to predict cell cycle on a continuum.</p>
<p>Our work demonstrated that peco is able to predict continuous cell cylce phase using a small set of cylcic genes: <em>CDK1</em>, <em>UBE2C</em>, <em>TOP2A</em>, <em>HISTH1E</em>, and <em>HISTH1C</em> (identified as cell cycle marker genes in studies of yeast (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC25624">Spellman et al., 1998</a>) and HeLa cells (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC117619">Whitfield et al., 2002</a>)).</p>
<p>Below we provide two use cases. Vignette 1 shows how to use the built-training dataset to predict continuous cell cycle. Vignette 2 shows how to make a training datast and build a predictor using training data.</p>
<p>Users can also view the vigenettes via <code><a href="https://rdrr.io/r/utils/browseVignettes.html">browseVignettes("peco")</a></code>.</p>
</div>
<div id="about-the-training-dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#about-the-training-dataset" class="anchor"></a>About the training dataset</h2>
<p><code>training_human</code> stores built-in training data of 101 significant cyclic genes. Below are the slots contained in <code>training_human</code>:</p>
<ul>
<li>
<code>predict.yy</code>: a gene by sample matrix (101 by 888) that stores predict cyclic expression values.</li>
<li>
<code>cellcycle_peco_reordered</code>: cell cycle phase in a unit circle (angle), ordered from 0 to 2<span class="math inline">\(pi\)</span>
</li>
<li>
<code>cellcycle_function</code>: lists of 101 function corresponding to the top 101 cyclic genes identified in our dataset</li>
<li>
<code>sigma</code>: standard error associated with cyclic trends of gene expression</li>
<li>
<code>pve</code>: proportion of variance explained by the cyclic trend</li>
</ul>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"training_human"</span>)</pre></body></html></div>
</div>
<div id="predict-cell-cycle-phase-using-gene-expression-data" class="section level2">
<h2 class="hasAnchor">
<a href="#predict-cell-cycle-phase-using-gene-expression-data" class="anchor"></a>Predict cell cycle phase using gene expression data</h2>
<p><code>peco</code> is integrated with <code>SingleCellExperiment</code> object in Bioconductor. Below shows an example of inputting <code>SingleCellExperiment</code> object to perform cell cycle phase prediction.</p>
<p><code>sce_top101genes</code> includes 101 genes and 888 single-cell samples and one assay slot of <code>counts</code>.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"sce_top101genes"</span>)
<span class="fu">assays</span>(<span class="no">sce_top101genes</span>)
<span class="co">#&gt; List of length 1</span>
<span class="co">#&gt; names(1): counts</span></pre></body></html></div>
<p>Transform the expression values to quantile-normalizesd counts-per-million values. <code>peco</code> uses the <code>cpm_quantNormed</code> slot as input data for predictions.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="no">sce_top101genes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/data_transform_quantile.html">data_transform_quantile</a></span>(<span class="no">sce_top101genes</span>)
<span class="co">#&gt; computing with 2 threads</span>
<span class="fu">assays</span>(<span class="no">sce_top101genes</span>)
<span class="co">#&gt; List of length 3</span>
<span class="co">#&gt; names(3): counts cpm cpm_quantNormed</span></pre></body></html></div>
<p>Apply the prediction model using function <code>cycle_npreg_outsample</code> and generate prediction results contained in a list object <code>pred_top101genes</code>.</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="no">pred_top101genes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/cycle_npreg_outsample.html">cycle_npreg_outsample</a></span>(
    <span class="kw">Y_test</span><span class="kw">=</span><span class="no">sce_top101genes</span>,
    <span class="kw">sigma_est</span><span class="kw">=</span><span class="no">training_human</span>$<span class="no">sigma</span>[<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">sce_top101genes</span>),],
    <span class="kw">funs_est</span><span class="kw">=</span><span class="no">training_human</span>$<span class="no">cellcycle_function</span>[<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">sce_top101genes</span>)],
    <span class="kw">method.trend</span><span class="kw">=</span><span class="st">"trendfilter"</span>,
    <span class="kw">ncores</span><span class="kw">=</span><span class="fl">1</span>,
    <span class="kw">get_trend_estimates</span><span class="kw">=</span><span class="fl">FALSE</span>)</pre></body></html></div>
<p>The <code>pred_top101genes$Y</code> contains a SingleCellExperiment object with the predict cell cycle phase in the <code>colData</code> slot.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">colData</span>(<span class="no">pred_top101genes</span>$<span class="no">Y</span>)$<span class="no">cellcycle_peco</span>)
<span class="co">#&gt; 20170905-A01 20170905-A02 20170905-A03 20170905-A06 20170905-A07 20170905-A08 </span>
<span class="co">#&gt;     1.099557     4.492477     2.607522     4.303982     4.052655     1.413717</span></pre></body></html></div>
<p>Visualize results of prediction for one gene. Below we choose CDK1 (“ENSG00000170312”). Because CDK1 is a known cell cycle gene, this visualization serves as a sanity check for the results of fitting. The fitted function <code>training_human$cellcycle_function[[1]]</code> was obtained from our training data.</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">y</span><span class="kw">=</span><span class="fu">assay</span>(<span class="no">pred_top101genes</span>$<span class="no">Y</span>,<span class="st">"cpm_quantNormed"</span>)[<span class="st">"ENSG00000170312"</span>,],
     <span class="kw">x</span><span class="kw">=</span><span class="fu">colData</span>(<span class="no">pred_top101genes</span>$<span class="no">Y</span>)$<span class="no">theta_shifted</span>, <span class="kw">main</span> <span class="kw">=</span> <span class="st">"CDK1"</span>,
     <span class="kw">ylab</span> <span class="kw">=</span> <span class="st">"quantile normalized expression"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="kw">y</span><span class="kw">=</span><span class="no">training_human</span>$<span class="no">cellcycle_function</span><span class="kw">[[</span><span class="st">"ENSG00000170312"</span>]](<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">0</span>,<span class="fl">2</span>*<span class="no">pi</span>, <span class="kw">length.out</span><span class="kw">=</span><span class="fl">100</span>)),
       <span class="kw">x</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">0</span>,<span class="fl">2</span>*<span class="no">pi</span>, <span class="kw">length.out</span><span class="kw">=</span><span class="fl">100</span>), <span class="kw">col</span> <span class="kw">=</span> <span class="st">"blue"</span>, <span class="kw">pch</span> <span class="kw">=</span><span class="fl">16</span>)</pre></body></html></div>
<p><img src="peco_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
<div id="visualize-cyclic-expression-trend-based-on-predicted-phase" class="section level2">
<h2 class="hasAnchor">
<a href="#visualize-cyclic-expression-trend-based-on-predicted-phase" class="anchor"></a>Visualize cyclic expression trend based on predicted phase</h2>
<p>Visualize results of prediction for the top 10 genesone genes. Use <code>fit_cyclical_many</code> to estimate cyclic function based on the input data.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="co"># predicted cell time in the input data</span>
<span class="no">theta_predict</span> <span class="kw">=</span> <span class="fu">colData</span>(<span class="no">pred_top101genes</span>$<span class="no">Y</span>)$<span class="no">cellcycle_peco</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">theta_predict</span>) <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="fu">colData</span>(<span class="no">pred_top101genes</span>$<span class="no">Y</span>))
<span class="co"># expression values of 10 genes in the input data</span>
<span class="no">yy_input</span> <span class="kw">=</span> <span class="fu">assay</span>(<span class="no">pred_top101genes</span>$<span class="no">Y</span>,<span class="st">"cpm_quantNormed"</span>)[<span class="fl">1</span>:<span class="fl">6</span>,]
<span class="co"># apply trendfilter to estimate cyclic gene expression trend</span>
<span class="no">fit_cyclic</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/fit_cyclical_many.html">fit_cyclical_many</a></span>(<span class="kw">Y</span><span class="kw">=</span><span class="no">yy_input</span>,
                                <span class="kw">theta</span><span class="kw">=</span><span class="no">theta_predict</span>)
<span class="co">#&gt; computing with 2 threads</span>
<span class="no">gene_symbols</span> <span class="kw">=</span> <span class="fu">rowData</span>(<span class="no">pred_top101genes</span>$<span class="no">Y</span>)$<span class="no">hgnc</span>[<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">yy_input</span>)]
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="kw">mfrow</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">2</span>,<span class="fl">3</span>))
<span class="kw">for</span> (<span class="no">i</span> <span class="kw">in</span> <span class="fl">1</span>:<span class="fl">6</span>) {
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw">y</span><span class="kw">=</span><span class="no">yy_input</span>[<span class="no">i</span>,],
     <span class="kw">x</span><span class="kw">=</span><span class="no">fit_cyclic</span>$<span class="no">cellcycle_peco_ordered</span>,
     <span class="kw">main</span> <span class="kw">=</span> <span class="no">gene_symbols</span>[<span class="no">i</span>],
     <span class="kw">ylab</span> <span class="kw">=</span> <span class="st">"quantile normalized expression"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/points.html">points</a></span>(<span class="kw">y</span><span class="kw">=</span><span class="no">fit_cyclic</span>$<span class="no">cellcycle_function</span><span class="kw">[[</span><span class="no">i</span>]](<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">0</span>,<span class="fl">2</span>*<span class="no">pi</span>, <span class="kw">length.out</span><span class="kw">=</span><span class="fl">100</span>)),
       <span class="kw">x</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">0</span>,<span class="fl">2</span>*<span class="no">pi</span>, <span class="kw">length.out</span><span class="kw">=</span><span class="fl">100</span>), <span class="kw">col</span> <span class="kw">=</span> <span class="st">"blue"</span>, <span class="kw">pch</span> <span class="kw">=</span><span class="fl">16</span>)
}</pre></body></html></div>
<p><img src="peco_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session information</h2>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()
<span class="co">#&gt; R version 3.6.2 (2019-12-12)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin15.6.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Catalina 10.15.7</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] peco_1.1.12                 doParallel_1.0.15          </span>
<span class="co">#&gt;  [3] iterators_1.0.12            foreach_1.4.7              </span>
<span class="co">#&gt;  [5] SingleCellExperiment_1.6.0  SummarizedExperiment_1.16.1</span>
<span class="co">#&gt;  [7] DelayedArray_0.12.3         BiocParallel_1.18.1        </span>
<span class="co">#&gt;  [9] matrixStats_0.56.0          Biobase_2.46.0             </span>
<span class="co">#&gt; [11] GenomicRanges_1.38.0        GenomeInfoDb_1.22.1        </span>
<span class="co">#&gt; [13] IRanges_2.20.2              S4Vectors_0.24.4           </span>
<span class="co">#&gt; [15] BiocGenerics_0.32.0        </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] viridis_0.5.1            BiocSingular_1.0.0       genlasso_1.4            </span>
<span class="co">#&gt;  [4] viridisLite_0.3.0        DelayedMatrixStats_1.6.1 assertthat_0.2.1        </span>
<span class="co">#&gt;  [7] GenomeInfoDbData_1.2.2   vipor_0.4.5              yaml_2.2.0              </span>
<span class="co">#&gt; [10] pillar_1.4.3             backports_1.1.5          lattice_0.20-38         </span>
<span class="co">#&gt; [13] glue_1.3.1               digest_0.6.23            XVector_0.26.0          </span>
<span class="co">#&gt; [16] colorspace_1.4-1         htmltools_0.4.0          Matrix_1.2-18           </span>
<span class="co">#&gt; [19] pkgconfig_2.0.3          zlibbioc_1.32.0          purrr_0.3.3             </span>
<span class="co">#&gt; [22] mvtnorm_1.0-11           scales_1.1.0             pracma_2.2.9            </span>
<span class="co">#&gt; [25] tibble_2.1.3             ggplot2_3.3.0            conicfit_1.0.4          </span>
<span class="co">#&gt; [28] magrittr_1.5             crayon_1.3.4             memoise_1.1.0           </span>
<span class="co">#&gt; [31] evaluate_0.14            fs_1.3.1                 MASS_7.3-51.4           </span>
<span class="co">#&gt; [34] beeswarm_0.2.3           geigen_2.3               tools_3.6.2             </span>
<span class="co">#&gt; [37] scater_1.12.2            lifecycle_0.1.0          stringr_1.4.0           </span>
<span class="co">#&gt; [40] munsell_0.5.0            irlba_2.3.3              compiler_3.6.2          </span>
<span class="co">#&gt; [43] pkgdown_1.5.1            rsvd_1.0.2               rlang_0.4.5             </span>
<span class="co">#&gt; [46] grid_3.6.2               RCurl_1.98-1.2           BiocNeighbors_1.2.0     </span>
<span class="co">#&gt; [49] circular_0.4-93          igraph_1.2.5             bitops_1.0-6            </span>
<span class="co">#&gt; [52] rmarkdown_2.3            boot_1.3-23              gtable_0.3.0            </span>
<span class="co">#&gt; [55] codetools_0.2-16         R6_2.4.1                 gridExtra_2.3           </span>
<span class="co">#&gt; [58] knitr_1.26               dplyr_0.8.3              rprojroot_1.3-2         </span>
<span class="co">#&gt; [61] desc_1.2.0               stringi_1.4.3            ggbeeswarm_0.6.0        </span>
<span class="co">#&gt; [64] Rcpp_1.0.5               tidyselect_0.2.5         xfun_0.11</span></pre></body></html></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Chiaowen Joyce Hsiao, Matthew Stephens.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
