<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Predict test-sample ordering using training labels (no update) — cycle_npreg_outsample • peco</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Predict test-sample ordering using training labels (no update) — cycle_npreg_outsample" />
<meta property="og:description" content="Apply the estimates of cycle_npreg_insample to another
gene expression dataset to infer an angle or cell cycle phase for
each cell." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">peco</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.15</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">Home</a>
</li>
<li>
  <a href="../articles/index.html">Vignettes</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jhsiao999/peco">Source</a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Predict test-sample ordering using training labels (no update)</h1>
    <small class="dont-index">Source: <a href='https://github.com/jhsiao999/peco/blob/master/R/cycle_npreg.R'><code>R/cycle_npreg.R</code></a></small>
    <div class="hidden name"><code>cycle_npreg_outsample.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Apply the estimates of cycle_npreg_insample to another
gene expression dataset to infer an angle or cell cycle phase for
each cell.</p>
    </div>

    <pre class="usage"><span class='fu'>cycle_npreg_outsample</span>(
  <span class='no'>Y_test</span>,
  <span class='no'>sigma_est</span>,
  <span class='no'>funs_est</span>,
  <span class='kw'>method.trend</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"trendfilter"</span>, <span class='st'>"loess"</span>, <span class='st'>"bspline"</span>),
  <span class='kw'>normed</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>polyorder</span> <span class='kw'>=</span> <span class='fl'>2</span>,
  <span class='kw'>method.grid</span> <span class='kw'>=</span> <span class='st'>"uniform"</span>,
  <span class='kw'>ncores</span> <span class='kw'>=</span> <span class='fl'>2</span>,
  <span class='kw'>grids</span> <span class='kw'>=</span> <span class='fl'>100</span>,
  <span class='kw'>get_trend_estimates</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>Y_test</th>
      <td><p>A SingleCellExperiment object.</p></td>
    </tr>
    <tr>
      <th>sigma_est</th>
      <td><p>Input from training data. A vector of gene-specific
standard error of the cyclic trends.</p></td>
    </tr>
    <tr>
      <th>funs_est</th>
      <td><p>Input fron training data. A vector of cyclic functions
estimating cyclic trends.</p></td>
    </tr>
    <tr>
      <th>method.trend</th>
      <td><p>Varous methods that can be applied to estimate
cyclic trend of gene expression levels.</p></td>
    </tr>
    <tr>
      <th>normed</th>
      <td><p>Is the data already normalized? TRUE or FALSE.</p></td>
    </tr>
    <tr>
      <th>polyorder</th>
      <td><p>We estimate cyclic trends of gene expression levels using
nonparamtric trend filtering. The default fits second degree polynomials.</p></td>
    </tr>
    <tr>
      <th>method.grid</th>
      <td><p>Method for defining bins along the circle.</p></td>
    </tr>
    <tr>
      <th>ncores</th>
      <td><p>We use doParallel package for parallel computing.</p></td>
    </tr>
    <tr>
      <th>grids</th>
      <td><p>number of bins to be selected along 0 to 2pi.</p></td>
    </tr>
    <tr>
      <th>get_trend_estimates</th>
      <td><p>To re-estimate the cylic trend based on
the predicted cell cycle phase or not (T or F). Default FALSE. This step
calls trendfilter and is computationally intensive.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A list with the following elements:</p>
<dt>Y</dt><dd><p>The input gene expression marix.</p></dd>
<dt>cell_times_est</dt><dd><p>Inferred angles or cell cycle phases, NOT
ordered.</p></dd>
<dt>loglik_est</dt><dd><p>Log-likelihood estimates for each gene.</p></dd>
<dt>cell_times_reordered</dt><dd><p>The inferred angles reordered (in
ascending order).</p></dd>
<dt>Y_reorded</dt><dd><p>The input gene expression matrix reordered by
cell_times_reordered.</p></dd>
<dt>sigma_reordered</dt><dd><p>Estimated standard error of the cyclic trend
for each gene, reordered by cell_times_reordered.</p></dd>
<dt>funs_reordered</dt><dd><p>A list of functions for approximating the
cyclic trends of gene express levels for each gene, reordered by
cell_times_reordered.</p></dd>
<dt>mu_reordered</dt><dd><p>Estimated cyclic trend of gene expression
values for each gene, reordered by cell_times_reordered.</p></dd>
<dt>prob_per_cell_by_celltimes</dt><dd><p>Probabilities of each cell belong
to each bin.</p></dd>

    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='cycle_npreg_insample.html'>cycle_npreg_insample</a></code> for obtaining parameteres for
    cyclic functions from training data,
    <code><a href='cycle_npreg_loglik.html'>cycle_npreg_loglik</a></code> for log-likehood at
    angles between 0 to 2pi,
    <code><a href='initialize_grids.html'>initialize_grids</a></code> for selecting
    angles in <code><a href='cycle_npreg_loglik.html'>cycle_npreg_loglik</a></code>,
    <code><a href='cycle_npreg_mstep.html'>cycle_npreg_mstep</a></code> for estimating cyclic functions given
    inferred phases from  <code><a href='cycle_npreg_loglik.html'>cycle_npreg_loglik</a></code></p>
<p>Other peco classifier functions: 
<code><a href='cycle_npreg_insample.html'>cycle_npreg_insample</a>()</code>,
<code><a href='cycle_npreg_mstep.html'>cycle_npreg_mstep</a>()</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='no'>SingleCellExperiment</span>)</div><div class='output co'>#&gt; <span class='message'>Loading required package: SummarizedExperiment</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomicRanges</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: stats4</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: BiocGenerics</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: parallel</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘BiocGenerics’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:parallel’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
#&gt; <span class='message'>    clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
#&gt; <span class='message'>    parLapplyLB, parRapply, parSapply, parSapplyLB</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:stats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    IQR, mad, sd, var, xtabs</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
#&gt; <span class='message'>    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
#&gt; <span class='message'>    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
#&gt; <span class='message'>    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
#&gt; <span class='message'>    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
#&gt; <span class='message'>    union, unique, unsplit, which, which.max, which.min</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: S4Vectors</span></div><div class='output co'>#&gt; <span class='warning'>Warning: package ‘S4Vectors’ was built under R version 3.6.3</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘S4Vectors’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    expand.grid</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: IRanges</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomeInfoDb</span></div><div class='output co'>#&gt; <span class='warning'>Warning: package ‘GenomeInfoDb’ was built under R version 3.6.3</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: Biobase</span></div><div class='output co'>#&gt; <span class='message'>Welcome to Bioconductor</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    Vignettes contain introductory material; view with</span>
#&gt; <span class='message'>    'browseVignettes()'. To cite Bioconductor, see</span>
#&gt; <span class='message'>    'citation("Biobase")', and for packages 'citation("pkgname")'.</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: DelayedArray</span></div><div class='output co'>#&gt; <span class='warning'>Warning: package ‘DelayedArray’ was built under R version 3.6.3</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: matrixStats</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘matrixStats’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:Biobase’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyMissing, rowMedians</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: BiocParallel</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘DelayedArray’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:matrixStats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    aperm, apply, rowsum</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='no'>sce_top101genes</span>)

<span class='co'># select top 5 cyclic genes</span>
<span class='no'>sce_top5</span> <span class='kw'>&lt;-</span> <span class='no'>sce_top101genes</span>[<span class='fu'><a href='https://rdrr.io/r/base/order.html'>order</a></span>(<span class='fu'>rowData</span>(<span class='no'>sce_top101genes</span>)$<span class='no'>pve_fucci</span>,
                                  <span class='kw'>decreasing</span><span class='kw'>=</span><span class='fl'>TRUE</span>)[<span class='fl'>1</span>:<span class='fl'>5</span>],]

<span class='co'># Select samples from NA18511 for our prediction example</span>
<span class='no'>coldata</span> <span class='kw'>&lt;-</span> <span class='fu'>colData</span>(<span class='no'>sce_top5</span>)
<span class='no'>which_samples_train</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span>(<span class='no'>coldata</span>)[<span class='no'>coldata</span>$<span class='no'>chip_id</span> <span class='kw'>!=</span> <span class='st'>"NA18511"</span>]
<span class='no'>which_samples_predict</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span>(<span class='no'>coldata</span>)[<span class='no'>coldata</span>$<span class='no'>chip_id</span> <span class='kw'>==</span> <span class='st'>"NA18511"</span>]

<span class='co'># learning cyclic functions of the genes using our training data</span>
<span class='no'>sce_top5</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='data_transform_quantile.html'>data_transform_quantile</a></span>(<span class='no'>sce_top5</span>)</div><div class='output co'>#&gt; <span class='message'>computing with 2 threads</span></div><div class='input'><span class='no'>expr_quant</span> <span class='kw'>&lt;-</span> <span class='fu'>assay</span>(<span class='no'>sce_top5</span>, <span class='st'>"cpm_quantNormed"</span>)
<span class='no'>Y_train</span> <span class='kw'>&lt;-</span> <span class='no'>expr_quant</span>[, <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span>(<span class='no'>expr_quant</span>) <span class='kw'>%in%</span> <span class='no'>which_samples_train</span>]
<span class='no'>theta_train</span> <span class='kw'>&lt;-</span>
    <span class='no'>coldata</span>$<span class='no'>theta_shifted</span>[<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span>(<span class='no'>coldata</span>) <span class='kw'>%in%</span> <span class='no'>which_samples_train</span>]
<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='no'>theta_train</span>) <span class='kw'>&lt;-</span>
    <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span>(<span class='no'>coldata</span>)[<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span>(<span class='no'>coldata</span>) <span class='kw'>%in%</span> <span class='no'>which_samples_train</span>]

<span class='co'># obtain cyclic function estimates</span>
<span class='no'>model_5genes_train</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='cycle_npreg_insample.html'>cycle_npreg_insample</a></span>(<span class='kw'>Y</span> <span class='kw'>=</span> <span class='no'>Y_train</span>,
                                           <span class='kw'>theta</span> <span class='kw'>=</span> <span class='no'>theta_train</span>,
                                           <span class='kw'>polyorder</span><span class='kw'>=</span><span class='fl'>2</span>,
                                           <span class='kw'>ncores</span><span class='kw'>=</span><span class='fl'>2</span>,
                                           <span class='kw'>method.trend</span><span class='kw'>=</span><span class='st'>"trendfilter"</span>)</div><div class='output co'>#&gt; <span class='message'>computing on 2 cores</span></div><div class='input'>
<span class='co'># predict cell cycle</span>
<span class='no'>model_5genes_predict</span> <span class='kw'>&lt;-</span> <span class='fu'>cycle_npreg_outsample</span>(
  <span class='kw'>Y_test</span><span class='kw'>=</span><span class='no'>sce_top5</span>[,<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span>(<span class='no'>sce_top5</span>) <span class='kw'>%in%</span> <span class='no'>which_samples_predict</span>],
  <span class='kw'>sigma_est</span><span class='kw'>=</span><span class='no'>model_5genes_train</span>$<span class='no'>sigma_est</span>,
  <span class='kw'>funs_est</span><span class='kw'>=</span><span class='no'>model_5genes_train</span>$<span class='no'>funs_est</span>,
  <span class='kw'>method.trend</span><span class='kw'>=</span><span class='st'>"trendfilter"</span>,
  <span class='kw'>ncores</span><span class='kw'>=</span><span class='fl'>2</span>,
  <span class='kw'>get_trend_estimates</span><span class='kw'>=</span><span class='fl'>FALSE</span>)

<span class='co'># estimate cyclic gene expression levels given cell cycle for each gene</span>
<span class='no'>predict_cyclic</span> <span class='kw'>&lt;-</span>
    <span class='fu'><a href='fit_cyclical_many.html'>fit_cyclical_many</a></span>(<span class='kw'>Y</span><span class='kw'>=</span><span class='fu'>assay</span>(<span class='no'>model_5genes_predict</span>$<span class='no'>Y</span>,<span class='st'>"cpm_quantNormed"</span>),
                      <span class='kw'>theta</span><span class='kw'>=</span><span class='fu'>colData</span>(<span class='no'>model_5genes_predict</span>$<span class='no'>Y</span>)$<span class='no'>cellcycle_peco</span>)</div><div class='output co'>#&gt; <span class='message'>computing with 2 threads</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/all.equal.html'>all.equal</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span>(<span class='no'>predict_cyclic</span><span class='kw'>[[</span><span class='fl'>2</span>]]), <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span>(<span class='no'>predict_cyclic</span><span class='kw'>[[</span><span class='fl'>1</span>]]))</div><div class='output co'>#&gt; [1] TRUE</div><div class='input'>
<span class='fu'><a href='https://rdrr.io/r/graphics/par.html'>par</a></span>(<span class='kw'>mfrow</span><span class='kw'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>2</span>,<span class='fl'>3</span>), <span class='kw'>mar</span><span class='kw'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>4</span>,<span class='fl'>4</span>,<span class='fl'>3</span>,<span class='fl'>1</span>))
<span class='kw'>for</span> (<span class='no'>g</span> <span class='kw'>in</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq_along</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span>(<span class='no'>model_5genes_predict</span>$<span class='no'>Y</span>))) {
  <span class='fu'><a href='https://rdrr.io/r/graphics/plot.html'>plot</a></span>(<span class='fu'>assay</span>(<span class='no'>model_5genes_predict</span>$<span class='no'>Y</span>,<span class='st'>"cpm_quantNormed"</span>)[
      <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span>(<span class='no'>model_5genes_predict</span>$<span class='no'>Y</span>)[<span class='no'>g</span>],],
       <span class='kw'>x</span><span class='kw'>=</span><span class='fu'>colData</span>(<span class='no'>model_5genes_predict</span>$<span class='no'>Y</span>)$<span class='no'>cellcycle_peco</span>, <span class='kw'>axes</span><span class='kw'>=</span><span class='fl'>FALSE</span>,
       <span class='kw'>xlab</span><span class='kw'>=</span><span class='st'>"FUCCI phase"</span>,
       <span class='kw'>ylab</span><span class='kw'>=</span><span class='st'>"Predicted phase"</span>)
  <span class='fu'><a href='https://rdrr.io/r/graphics/points.html'>points</a></span>(<span class='kw'>y</span><span class='kw'>=</span><span class='no'>predict_cyclic</span>$<span class='no'>cellcycle_function</span><span class='kw'>[[</span>
           <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span>(<span class='no'>model_5genes_predict</span>$<span class='no'>Y</span>)[<span class='no'>g</span>]]](
    <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>0</span>, <span class='fl'>2</span>*<span class='no'>pi</span>, <span class='kw'>length.out</span> <span class='kw'>=</span> <span class='fl'>100</span>)),
    <span class='kw'>x</span><span class='kw'>=</span><span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span>(<span class='fl'>0</span>, <span class='fl'>2</span>*<span class='no'>pi</span>, <span class='kw'>length.out</span> <span class='kw'>=</span> <span class='fl'>100</span>),
    <span class='kw'>pch</span><span class='kw'>=</span><span class='fl'>16</span>, <span class='kw'>col</span><span class='kw'>=</span><span class='st'>"royalblue"</span>)
  <span class='fu'><a href='https://rdrr.io/r/graphics/axis.html'>axis</a></span>(<span class='fl'>2</span>); <span class='fu'><a href='https://rdrr.io/r/graphics/axis.html'>axis</a></span>(<span class='fl'>1</span>,<span class='kw'>at</span><span class='kw'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>0</span>,<span class='no'>pi</span>/<span class='fl'>2</span>, <span class='no'>pi</span>, <span class='fl'>3</span>*<span class='no'>pi</span>/<span class='fl'>2</span>, <span class='fl'>2</span>*<span class='no'>pi</span>),
                <span class='kw'>labels</span><span class='kw'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='fl'>0</span>,<span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span>(<span class='no'>pi</span>/<span class='fl'>2</span>), <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span>(<span class='no'>pi</span>),
                         <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span>(<span class='fl'>3</span>*<span class='no'>pi</span>/<span class='fl'>2</span>), <span class='fu'><a href='https://rdrr.io/r/base/expression.html'>expression</a></span>(<span class='fl'>2</span>*<span class='no'>pi</span>)))
  <span class='fu'><a href='https://rdrr.io/r/graphics/abline.html'>abline</a></span>(<span class='kw'>h</span><span class='kw'>=</span><span class='fl'>0</span>, <span class='kw'>lty</span><span class='kw'>=</span><span class='fl'>1</span>, <span class='kw'>col</span><span class='kw'>=</span><span class='st'>"black"</span>, <span class='kw'>lwd</span><span class='kw'>=</span><span class='fl'>.7</span>)
  <span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span>(<span class='no'>model_5genes_predict</span>$<span class='no'>Y_reordered</span>)[<span class='no'>g</span>])
}
<span class='fu'><a href='https://rdrr.io/r/graphics/title.html'>title</a></span>(<span class='st'>"Predicting cell cycle phase for NA18511"</span>, <span class='kw'>outer</span><span class='kw'>=</span><span class='fl'>TRUE</span>)</div><div class='img'><img src='cycle_npreg_outsample-1.png' alt='' width='700' height='433' /></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Chiaowen Joyce Hsiao, Matthew Stephens.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


